File: SmartG.Entities/Models/Post.cs
using System;
namespace SmartG.Entities.Models
{
    public class Post:BaseEntity
    {
        public Guid PostId { get; set; }
        public string? Title { get; set; }
        public string? Content { get; set; }
        public string? Excerpt { get; set; }
        public string? MetaDescription { get; set; }
        public string? MetaKeyWords { get; set; }
        public string? Slug { get; set; }


        public int? SgCategoryId { get; set; }
        public int? SgImageId { get; set; }
        public int? SgGalleryId { get; set; }
        public ICollection<Comment>? Comments { get; set; }


        public ICollection<ContentBlock>? ContentBlocks { get; set; }
        

    }
}

File: SmartG-client/src/app/_interfaces/post.ts
import { Category } from "./category";
import { ContentBlock } from "./content-block";
import { Gallery } from "./gallery";
import { Image } from "./image";

    export interface Post {
        postId: string;
        title: string;
        content: string;
        excerpt: string;
        metaDescription: string;
        metaKeyWords: string;
        slug: string;
        category: Category;
        imageId: number;

    galleryId:number;
        image: Image;
        contentBlocks: ContentBlock[];
        dateCreated: Date;
        dateUpdated: Date;
        deleted: boolean;
        authorId: string;
        gallery:Gallery;
    }

